### PR TITLE
feat(ui): Tweak AvatarList design and add to Issue Details

### PR DESCRIPTION
**Pull request by billyvg**
* Spread avatars out a bit more
* Move number to end of list and add a tooltip
* Add z-index on hover
* Change Issue Details to use AvatarList

screen shot 2018-05-16 at 7 17 43 pm

screen shot 2018-05-16 at 7 18 27 pm

screen shot 2018-05-16 at 7 18 09 pm

### CODE REVIEW

File: src/sentry/static/sentry/app/components/avatar/avatarList.jsx [hunk 1]
```diff
@@ -19,31 +22,63 @@ export default class AvatarList extends React.Component {
...
       <Flex direction="row-reverse">
-        {visibleUsers.map(user => {
-          return <StyledAvatar key={user.id} user={user} size={avatarSize} hasTooltip />;
-        })}
         {!!numCollapsedUsers && (
-          <CollapsedUsers size={avatarSize}>{numCollapsedUsers}</CollapsedUsers>
+          <Tooltip title={`${numCollapsedUsers} other users`}>
+            <CollapsedUsers size={avatarSize}>
+              <Plus>+</Plus>
+              {Math.max(numCollapsedUsers, 99)}
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/avatar/avatarList.jsx]: Is this something to worry about? Are there teams with > 99 (+maxVisible) users?

Comment (lynnagara) [reply src/sentry/static/sentry/app/components/avatar/avatarList.jsx]: I assume there's no limit, so setting a max of 99 could be incorrect. Where does the 99 number come from anyway?

Comment (billyvg) [reply src/sentry/static/sentry/app/components/avatar/avatarList.jsx]: Trying to prevent overflow when there's 3 digits. Maybe we can move the plus so that from 1-98 it's `+<number>` and then it becomes `99+`

Comment (lynnagara) [reply src/sentry/static/sentry/app/components/avatar/avatarList.jsx]: Doesn't the count always show 99 in that case?

Comment (billyvg) [reply src/sentry/static/sentry/app/components/avatar/avatarList.jsx]: oops yeah, should be min.

Comment (lynnagara) [reply src/sentry/static/sentry/app/components/avatar/avatarList.jsx]: Removing the plus would be fine

Comment (lynnagara) [reply src/sentry/static/sentry/app/components/avatar/avatarList.jsx]: I think you meant Max.min then?

Comment (lynnagara) [reply src/sentry/static/sentry/app/components/avatar/avatarList.jsx]: In any case, I think we should drop it and just remove the plus, or change the font size or something along those lines

File: src/sentry/static/sentry/app/views/groupDetails/seenBy.jsx [hunk 2]
```diff
@@ -31,37 +32,50 @@ const GroupSeenBy = createReactClass({
...
-            <Tooltip title={t("People who've viewed this issue")}>
-              <span className="icon-eye" />
-            </Tooltip>
-          </li>
-          {seenByNodes}
-        </ul>
-      </div>
+      <SeenByWrapper className="seen-by">
+        <AvatarList
+          users={seenBy.filter(user => true || activeUser.id !== user.id)}
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/groupDetails/seenBy.jsx]: is this supposed to filter something?

File: src/sentry/static/sentry/app/views/groupDetails/seenBy.jsx [hunk 3]
```diff
@@ -31,37 +32,50 @@ const GroupSeenBy = createReactClass({
...
+          avatarSize={28}
+          maxVisibleAvatars={10}
+          tooltipOptions={{html: true}}
+          renderTooltip={user =>
+            `${_.escape(userDisplayName(user))} <br/> ${_.escape(
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/groupDetails/seenBy.jsx]: Could we drop `_.escape()`?

File: src/sentry/static/sentry/app/views/groupDetails/seenBy.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/groupDetails/seenBy.jsx]: The `true ||` was leftover due to debugging :/ need to make a lint rule for that.

It filters out the current user from the "Seen by" list.

File: src/sentry/static/sentry/app/views/groupDetails/seenBy.jsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/groupDetails/seenBy.jsx]: We could drop it for the date, but since user display name takes in user input, and we are displaying raw html in the tooltip, we should keep `escape` to protect against XSS.

File: src/sentry/static/sentry/app/views/groupDetails/seenBy.jsx [hunk 4]
```diff
@@ -31,37 +32,48 @@ const GroupSeenBy = createReactClass({
...
-            <Tooltip title={t("People who've viewed this issue")}>
-              <span className="icon-eye" />
-            </Tooltip>
-          </li>
-          {seenByNodes}
-        </ul>
-      </div>
+      <SeenByWrapper className="seen-by">
+        <AvatarList
+          users={seenBy.filter(user => activeUser.id != user.id)}
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/groupDetails/seenBy.jsx]: Why did you change it from `!==` to `!=`?

**Review by lynnagara — APPROVED** [review]
